src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -4763,24 +4771,31 @@ class StubGenerator: public StubCodeGenerator {
       StubRoutines::_base64_encodeBlock = generate_base64_encodeBlock();
     }
 #endif
+#endif // COMPILER2_OR_JVMCI
   }
 
  public:
-  StubGenerator(CodeBuffer* code, int phase) : StubCodeGenerator(code) {
-    if (phase == 0) {
-      generate_initial();
-    } else if (phase == 1) {
-      generate_phase1(); // stubs that must be available for the interpreter
-    } else {
-      generate_all();
-    }
+  StubGenerator(CodeBuffer* code, StubsKind kind) : StubCodeGenerator(code) {
```

Comment:
The code is repeated on multiple platforms. It makes sense to lift it to `StubCodeGenerator`.

Reply:
It is because `StubGenerator` is declared and defined separately on each platform. Doing that here will add more complexity for already not small changes.  Filed new RFE: [JDK-8304750](https://bugs.openjdk.org/browse/JDK-8304750)